tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1418,6 +1419,53 @@ public function testCommentInTable(): void
         $table = $this->schemaManager->listTableDetails('table_with_comment');
         self::assertSame('Foo with control characters \'\\', $table->getComment());
     }
+
+    public function testCreatedCompositeForeignKeyOrderIsCorrectAfterCreation(): void
+    {
+        if (! $this->schemaManager->getDatabasePlatform()->supportsForeignKeyConstraints()) {
+            self::markTestSkipped('Platform does not support foreign keys.');
+        }
+
+        $foreignKey                = 'fk_test_foreign_key';
+        $tableForeignKeyReferenced = 'test_table_foreign_key_referenced';
+        $tableWithForeignKey       = 'test_table_with_foreign_key';
+        $localColumns              = ['child_col2', 'child_col1'];
+        $foreignColumns            = ['col2', 'col1'];
+
+        $this->schemaManager->tryMethod('dropForeignKey', $foreignKey, $tableWithForeignKey);
```

Comment:
Dropping the FK shouldn't be necessary. Instead, the order of `DROP TABLE`'s should be changed: first the referencing, then the referenced.

Reply:
@morozov Ah, damn, of course it shouldn't matter as long as the order in which they are dropped is correct. 

I guess it is indeed late 😄 

Fixed this now and also shortened the names so the OCI tests are hopefully successful as well.

Reply:
btw. any reason why this line ` Type::addType('point', PointType::class); ` isn't something like:
`Type::hasType('point') ? Type::overrideType('point', PointType::class) : Type::addType('point', PointType::class);`

Reply:
It probably should be like that. Nobody bothered to fix it yet.